src/ORM/Table.php

```diff
@@ -1011,6 +1011,14 @@ public function findAll(Query $query, array $options)
      */
     public function findList(Query $query, array $options)
     {
+        if (!isset($options['keyField']) && !isset($options['valueField'])) {
+            $select = $query->clause('select');
+            if ($select && count($select) <= 2) {
+                $options['keyField'] = array_shift($select);
+                $options['valueField'] = array_shift($select) ?: $options['keyField'];
```

Comment:
It throws a `Undefined property: Cake\Test\TestCase\ORM\TableTest::$Users` notice, which at least tells you something, that this aint working. You will then at some point at least realize that you need to add keyField/valueField.
Or we could extract the model alias here and throw it away for the definition setting.
If that makes more sense.